### DISCUSSION TITLE
Updates for private GitHub-hosted NPM packages not showing up

### DISCUSSION THREAD

**Discussion by RichiCoder1** [renovatebot/renovate#13873] [Request Help]
I noticed while making some renovate tweaks that private NPM packages were not longer receiving automatic updates. As far as I know this was working previously, and is no longer working now. I don't see any errors in the logs and the packages are showing up in the logs w/ requests to `npm.pkg.github.com`.

`.npmrc`

```ini
@uship:registry=https://npm.pkg.github.com/
```

renovate config (relevant section)
```json5
  encrypted: {
    npmToken: "**encyrpted**",
  },
  npmrc: "@uship:registry=https://npm.pkg.github.com/\n//npm.pkg.github.com/:_authToken=${NPM_TOKEN}",
  hostRules: [
    {
      matchHost: "npm.pkg.github.com",
      encrypted: {
        token: "**encrypted**"
      }
    },
    {
      matchHost: "nuget.pkg.github.com",
      encrypted: {
        token: "**encrypted**"
      }
    },
  ],
```

Edit:
I'm noticing it's missing some _non-private_ normal NPM packages too. I currently have `@middy/core: ^2.5.4` and renovate is not picking up the current version (as of writing) `2.5.7`.
![image](https://user-images.githubusercontent.com/2391878/151453689-a65b403b-8edb-4689-a62c-332700a92806.png)
![image](https://user-images.githubusercontent.com/2391878/151453705-48b00864-97f8-4a2c-8e8f-ba3dcc25f6a1.png)

![image](https://user-images.githubusercontent.com/2391878/151453761-a6b38f7f-d893-4eed-bc85-9059361faaac.png)

![image](https://user-images.githubusercontent.com/2391878/151453789-328879ad-7bc7-41c9-9261-7d4d5cbb2f48.png)

**Comment by rarkins** [comment]
Please read the documentation for rangeStrategy as that most likely explains things

**Reply by RichiCoder1** [reply]
Ah, that would explain the middy, but that doesn't explain the private packages which did have a `minor` version bump.

**Reply by rarkins** [reply]
You haven't provided enough information about the others

**Reply by RichiCoder1** [reply]
Here's the package in question:

![image](https://user-images.githubusercontent.com/2391878/151500023-ec9a6e9e-8d16-4b21-9c11-2870a805ebb3.png)
![image](https://user-images.githubusercontent.com/2391878/151500715-a0398be0-e1f1-42c7-a89a-707f0435b174.png)
![image](https://user-images.githubusercontent.com/2391878/151500924-867ab4fb-2a17-4c85-9c07-06015dd354e1.png)


As far as I'm aware, we're using the default `rangeStrategy` with `config:base` and `:preserveSemverRanges`

**Reply by RichiCoder1** [reply]
Apologies if that's also expected behavior

**Reply by RichiCoder1** [reply]
Also, if there's another diagnostic I can run, lmk

**Reply by rarkins** [reply]
It's expected behavior for the same reason as middy. If this repository is a library, set `rangeStrategy=update-lockfile` for npm. If it's an application, set `rangeStrategy=pin`.

**Reply by RichiCoder1** [reply]
Switching strategy did it, apologies! Our organization's defaults where a little over the place.
